src/values/Value.js

```diff
@@ -34,6 +34,7 @@ export default class Value {
     switch (typeName) {
       case "empty": return EmptyValue;
       case "undefined": return UndefinedValue;
```

Comment:
I removed the undefined case and it still passes all the test cases so it must have been extraneous. Good catch